client/client.go

```diff
@@ -1525,12 +1544,12 @@ OUTER:
 	}
 }
 
-// watchNodeUpdates periodically checks for changes to the node attributes or meta map
-func (c *Client) watchNodeUpdates() {
+// watchNodeUpdates periodically checks for changes to the node attributes or
+// meta map. The passed hashes are the initial hash values for the attribute and
+// metadata of the node respectively.
+func (c *Client) watchNodeUpdates(attrHash, metaHash uint64) {
 	c.logger.Printf("[DEBUG] client: periodically checking for node changes at duration %v", nodeUpdateRetryIntv)
 
-	// Initialize the hashes
-	_, attrHash, metaHash := c.hasNodeChanged(0, 0)
```

Comment:
On line 1557- don't overwrite attrHash and metaHash if it isn't necessary.

Reply:
It is necessary. The hasNodeChanged takes in the old and returns the new. If they differ it tells you whether there was an update. So you have to capture the changed hashes.